pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -161,20 +175,25 @@ func (r *RepoSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 
 	// Create secret in config-management-system namespace using the
 	// existing secret in the reposync.namespace.
-	if err := upsertSecrets(ctx, rs, r.client, reconcilerName); err != nil {
+	if sKey, err := upsertSecrets(ctx, rs, r.client, reconcilerKey); err != nil {
 		var authType configsync.AuthType
 		if rs.Spec.SourceType == string(v1beta1.GitSource) {
 			authType = rs.Spec.Auth
 		} else if rs.Spec.SourceType == string(v1beta1.HelmSource) {
 			authType = rs.Spec.Helm.Auth
 		}
-		log.Error(err, "RepoSync failed secret creation", "auth", authType)
+		log.Error(err, "Managed object upsert failed",
```

Comment:
Is the motivation for the extra return value only for this logging?

If so, perhaps we could push this log statement down into the `upsertSecrets` function so that we can maintain the logging granularity by secret name without needing the extra return value

Reply:
This seems like a nice follow up. I would want to do it for all the upsert functions, but I don't think it's worth holding up the rest of this refactor.

Reply:
ack